src/uiSelectController.js

```diff
@@ -454,9 +454,16 @@ uis.controller('uiSelectCtrl',
   };
 
   ctrl.clear = function($event) {
+    var locals = {};
+    locals[ctrl.parserResult.itemName] = ctrl.selected;
+
     ctrl.select(null);
     $event.stopPropagation();
     $timeout(function() {
+      ctrl.onRemoveCallback($scope, {
```

Comment:
@Jefiozie I see what you are saying. I might be overlooking something here but it seems to me that `uiSelectMultiple` has the `removeChoice` method on its controller and then the `uiSelectMatchDirective` view uses the `uiSelectMultiple` controller to call it like so `ng-click="$selectMultiple.removeChoice($index)"`. 

Currently `uiSingleSelect` doesn't have a controller so I would have to add the method directly to the scope of `uiSingleSelect` in the link function. Then I would call it from the `uiSelectMatchDirective` single view like so `ng-click="remove($event)"`. I just haven't seen too many methods bound directly to the scope in the library so I wasn't sure if this was correct. What do you think?